specification/threatmodel.adoc

```diff
@@ -1,187 +1,300 @@
 
-| TEE CPU State Protection | State Isolation | Required | Confidential qualifier and privilege levels M S HS U | Prevent untrusted code from arbitrarily accessing/modifying TEE CPU state | AP-TEE TG to specify 
+| Memory Confidentiality | Memory isolation (read) | Required | cryptography
+ and/or MMU, MPU, MTT extension | Prevent non-TCB components from reading
+TEE memory | Sv, Smmtt
 
-| Memory Confidentiality | Memory isolation (read)     | Required                | cryptography and/or MMU, MPU, PMA/MTT extension | Prevent untrusted components from reading TEE memory       | AP-TEE TG specify 
-| Memory Confidentiality | Cipher text read prevention | Required                | cryptography and/or MMU, MPU, PMA/MTT extension | Prevent untrusted code from accessing encrypted TEE memory | AP-TEE TG specify 
-| Memory Confidentiality | Per TEE encryption          | Implementation-specific | cryptography and/or MMU, MPU, PMA/MTT extension | Each VM has one or more unique keys                        | AP-TEE TG to recommend 
-| Memory Confidentiality | Memory encryption strength  | Implementation-specific | cryptography                                    | Encryption algorithm and key strength                      | AP-TEE TG to recommend 
-| Memory Confidentiality | Number of encryption keys   | Implementation-specific | cryptography                                    | Number of TEE keys supported                               | AP-TEE TG to recommend 
+| Memory Confidentiality | Cipher text read prevention | Required | cryptography
+ and/or MMU, MPU, MTT extension | Prevent non-TCB components from accessing
+encrypted TEE memory | Svpams (Supervisor Domains)
 
-| Memory Integrity | Memory integrity against SW attacks | Required                | MMU, MPU, PMA/MTT extension                     | Prevent SW attacks such as remapping aliasing replay corruption etc.                                      | AP-TEE TG to specify 
-| Memory Integrity | Memory integrity against HW attacks | Implementation-specific | cryptography and/or MMU, MPU, PMA/MTT extension | Prevent HW attacks DRAM-bus attacks and physical attacks that replace TEE memory with tampered / old data | AP-TEE TG to recommend 
-| Memory Integrity | Memory isolation (Write exec)       | Required                | cryptography and/or MMU, MPU, PMA/MTT extension | Prevent TEE from executing from normal memory; Enforce integrity of TEE data on writes                    | AP-TEE TG specify 
-| Memory Integrity | Rowhammer attack prevention         | Implementation-specific | cryptography and/or memory-specific extension   | Prevent untrusted code from flipping bits of TEE memory                                                   | AP-TEE TG to recommend 
+| Memory Confidentiality | Per TEE encryption | Implementation-specific |
+cryptography and/or MMU, MPU, MTT extension | Each VM has one or more
+unique keys  | Svpams (Supervisor Domains)
 
```

Comment:
Why is MMU/MPU/MTT not sufficient to provide memory confidentiality property? How does per-TEE encryption provide confidentity. Per TEE eencryption would still not allow letting one TEE access another TEEs ciphertext due to the ciphertext disclosure issue. By corollary if the memory region is not accessible in first place then fact that it is per-TEE encrypted or not may be moot?

Reply:
hence the "or" - this is against physical attacks - agree that it goes hand-in-hand with preventing ciphertext disclosure, as well as memory integrity and replay-protection. Adding a note to the effect (ptal at updated text).